doc/pandas-2.0/source/goals.rst

```diff
@@ -0,0 +1,194 @@
+.. _goals:
+
+=======================
+ Goals and Motivations
+=======================
+
+.. note::
+
+  These documents are largely written by Wes McKinney, and at this point
+  reflect his opinions for the time being. Many things may change as we discuss
+  and work to reach a consensus about the path forward.
+
+The pandas codebase is now over 8 years old, having grown to over 200,000 lines
+of code from its original ~10,000 LOC in the original 0.1 open source release
+in January 2010.
+
+At a high level, the "pandas 2.0" effort is based on a number of observations:
+
+* The pandas 0.x series of releases have consisted with huge amounts of
```

Comment:
Good points. I mainly fixated on `.ix` because the `.loc` and `.iloc` indexing operators will probably need to be reimplemented as part of this internals overhaul, and having to drag along things like `.ix` would add implementation burden for unclear benefit. I agree talking about other refactoring / cleaning is a distraction. Will make some amendments to make this more clear.

The other intent of this first point was that the iterative / agile development style of the project (from its early days until now) has made it difficult to consider large/invasive changes to the internals, and after so much time we are due to seriously contemplate what's working well and what's not working well.